fvm/evm/handler/blockHashList.go

```diff
@@ -147,19 +148,22 @@ func (bhl *BlockHashList) updateBlockHashAt(idx int, bh gethCommon.Hash) error {
 	// fetch the bucket
 	bucketNumber := idx / hashCountPerBucket
 	bucket, err := bhl.fetchBucket(bucketNumber)
+	cpy := make([]byte, len(bucket))
+	copy(cpy, bucket)
+
 	if err != nil {
 		return err
 	}
```

Comment:
```suggestion
	if err != nil {
		return err
	}
	cpy := make([]byte, len(bucket))
	copy(cpy, bucket)
```

Reply:
Good point, we can perform the bucket copy, after we know there's no error.